packages/wrangler/src/__tests__/patch-config.test.ts

```diff
@@ -0,0 +1,340 @@
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+
+				[[kv_namespaces]]
+				binding = "KV"
+
+				[[kv_namespaces]]
+				binding = "KV2"
+
+			`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name",
+					"kv_namespaces": [
+						{
+							"binding": "KV"
+						},
+						{
+							"binding": "KV2"
+						}
+					]
+				}
+			`,
+	},
+	{
+		name: "add a new D1 binding when only KV bindings exist",
+		original: {
+			kv_namespaces: [
+				{
+					binding: "KV",
+				},
+			],
+		},
+		patch: {
+			d1_databases: [
+				{
+					binding: "DB",
+				},
+			],
+		},
+		expectedToml: dedent`
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+
+				[[kv_namespaces]]
+				binding = "KV"
+
+				[[d1_databases]]
+				binding = "DB"
+
+			`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name",
+					"kv_namespaces": [
+						{
+							"binding": "KV"
+						}
+					],
+					"d1_databases": [
+						{
+							"binding": "DB"
+						}
+					]
+				}
+			`,
+	},
+	{
+		name: "add a new field",
+		original: {
+			kv_namespaces: [
+				{
+					binding: "KV",
+				},
+			],
+		},
+		patch: {
+			compatibility_flags: ["nodejs_compat"],
+		},
+		expectedToml: dedent`
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+				compatibility_flags = [ "nodejs_compat" ]
+
+				[[kv_namespaces]]
+				binding = "KV"
+
+			`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name",
+					"kv_namespaces": [
+						{
+							"binding": "KV"
+						}
+					],
+					"compatibility_flags": [
+						"nodejs_compat"
+					]
+				}
+			`,
+	},
+	{
+		name: "make multiple edits at the same time",
+		original: {
+			kv_namespaces: [
+				{
+					binding: "KV",
+				},
+			],
+			d1_databases: [
+				{
+					binding: "DB",
+				},
+			],
+		},
+		patch: {
+			kv_namespaces: [
+				{
+					binding: "KV2",
+				},
+				{
+					binding: "KV3",
+				},
+			],
+			d1_databases: [
+				{
+					binding: "DB2",
+				},
+			],
+		},
+		expectedToml: dedent`
+				compatibility_date = "2022-01-12"
+				name = "test-name"
+
+				[[kv_namespaces]]
+				binding = "KV"
+
+				[[kv_namespaces]]
+				binding = "KV2"
+
+				[[kv_namespaces]]
+				binding = "KV3"
+
+				[[d1_databases]]
+				binding = "DB"
+
+				[[d1_databases]]
+				binding = "DB2"
+
+			`,
+		expectedJson: dedent`
+				{
+					"compatibility_date": "2022-01-12",
+					"name": "test-name",
+					"kv_namespaces": [
+						{
+							"binding": "KV"
+						},
+						{
+							"binding": "KV2"
+						},
+						{
+							"binding": "KV3"
+						}
+					],
+					"d1_databases": [
+						{
+							"binding": "DB"
+						},
+						{
+							"binding": "DB2"
+						}
+					]
+				}
+			`,
+	},
+];
+
+describe("experimental_patchConfig()", () => {
+	runInTempDir();
+	describe.each(["json", "toml"])("%s", (configType) => {
+		it.each(testCases)(
```

Comment:
You might want to add one more test to check if the config file is not found

Reply:
made the configPath required now instead